image/preseed/preseed.go

```diff
@@ -39,9 +54,153 @@ type preseedOpts struct {
 	PreseedChrootDir string
 	SystemLabel      string
 	WritableDir      string
+	PreseedSignKey   string
 }
 
 type targetSnapdInfo struct {
 	path    string
 	version string
 }
+
+var (
+	getKeypairManager        = signtool.GetKeypairManager
+	newToolingStoreFromModel = tooling.NewToolingStoreFromModel
+	trusted                  = sysdb.Trusted()
+)
+
+func MockTrusted(mockTrusted []asserts.Assertion) (restore func()) {
+	prevTrusted := trusted
+	trusted = mockTrusted
+	return func() {
+		trusted = prevTrusted
+	}
+}
+
+func writePreseedAssertion(artifactDigest []byte, opts *preseedOpts) error {
+	keypairMgr, err := getKeypairManager()
+	if err != nil {
+		return err
+	}
+
+	key := opts.PreseedSignKey
+	if key == "" {
+		key = `default`
+	}
+	privKey, err := keypairMgr.GetByName(key)
+	if err != nil {
+		// TRANSLATORS: %q is the key name, %v the error message
+		return fmt.Errorf(i18n.G("cannot use %q key: %v"), key, err)
```

Comment:
This error uses translations, but other errors do not. Most likely the error from GetByName() aren't translated either.

Reply:
Indeed. For the time being I'll leave it as is,`cmd_sign.go` (that I based this on) has the same property.